src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -1745,6 +1762,17 @@ _mongoc_client_retryable_write_command_with_stream (
       mongoc_server_stream_cleanup (retry_server_stream);
    }
 
+   // If a retry attempt fails with an error labeled NoWritesPerformed,
+   // drivers MUST return the original error.
+   if (mongoc_error_has_label (reply, "NoWritesPerformed") &&
+       original_error.set) {
+      if (error) {
+         *error = original_error.error;
+      }
+      bson_copy_to (&original_error.reply, reply);
+      bson_destroy (&original_error.reply);
```

Comment:
Potential leak: `original_error.reply` must be destroyed if `original_error.set` is true even if `mongoc_error_has_label (reply, "NoWritesPerformed")` is false.